### ISSUE TITLE
Passing concatenated tuple to positional rest arg results in unknown argument type error

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#6458]
**Describe the bug**
Given the following code:

```python
from __future__ import annotations


def inner(*args: object) -> None:
    ...


def outer(*args: object) -> None:
    args = (*args, 'another arg')

    inner(*args)
```

An error is generated on the last line:

```
blah.py
  blah.py:11:5 - error: Argument type is unknown
    Argument corresponds to parameter "args" in function "inner" (reportUnknownArgumentType)
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool?
Command line

Which version?
1.1.336

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.337, which I just published. It will also be included in a future release of pylance.
